src/figuresClasses.ts

```diff
@@ -1,11 +1,89 @@
-export interface Figure {}
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+export interface Figure {
+  shape: string;
+  color: Color;
+  getArea(): number;
+}
 
-export class Triangle implements Figure {}
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sortSide = [this.a, this.b, this.c].sort((acc, prev) => acc - prev);
+
+    if (
+      a <= 0 ||
+      b <= 0 ||
+      c <= 0 ||
+      sortSide[2] >= sortSide[0] + sortSide[1]
+    ) {
+      throw new Error('incorrect sides');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const areaTriangle = Math.sqrt(
+      s * (s - this.a) * (s - this.b) * (s - this.c),
+    );
+
+    return +areaTriangle.toFixed(2);
+  }
+}
 
-export class Rectangle implements Figure {}
+export class Circle implements Figure {
+  shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error('incorrect radius');
```

Comment:
[CODE STYLE] - When throwing an error, try to provide a detailed description in the error message to make it clear what went wrong.

GOOD EXAMPLE:
throw new Error('Circle radius must be a positive number.');